### PR TITLE
Extract telescope positions from DB

### PR DESCRIPTION
**Pull request by GernotMaier**
Address selection of telescopes to form an array from the DB and add functionality to plot_array_layout (plot_array_layout is a great application to test this). Reduce dependency on ArrayLayout class (which should be used only for coordinate transformations).

Improved functionality for ArrayModel in `array_element` selections to describe an array:

1. use a list of telescopes in a ecsv file (as before)
2. use a pre-defined array layout name; layouts are defined in the DB in the parameter `array_layouts` stored in the `sites collection (new); see [gitlab merge request](https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/model_parameters/-/merge_requests/37#f533e96db03c395f9acbb50883fcaa8c3cbe9709)
3. provide a list of telescope on the command line (new)

For 2.: allowed is e.g., `LSTN-01 LSTN-02 MSTN`, meaning 2 LSTs plus all northern MSTs. This required to add a function `db_handler.get_available_array_elements_of_type`

Improved additionally documentation. Note that in all cases changes are applied to the modules touched in this PR to pass the "D" (documentation) checks by ruff.

Remove dependency on array_layout model for plot_array_layout and make_regular_arrays.

Remove `names.validate_array_layout_name()`: this is not needed anymore as the call to the DB with throw an error when an array layout with a given name is not found in the DB.

Note: a lot of changes are from the renaming of [Test-Layout, test-layout, TestLayout, testlayout] to `test_layout` (names.py mapped all in the brackets to the same name; which confused me!). A number of file name changes are from changes in the automatic file name generation in plot_array.py

Added a statement to suppress debug messages from mongoDB: this is useful, as it is very hard to debug issue especially in tests; mongoDB is printing a lot of debug statements which makes it hard to impossible to debug the actually issues.

### CODE REVIEW

File: simtools/applications/plot_array_layout.py [hunk 1]
```diff
@@ -122,106 +182,267 @@ def _get_site_from_telescope_list_name(telescope_list_file):
...
+    """
+    Read array layout positions from file(s) and return a list of layouts.
+
+    Parameters
+    ----------
+    args_dict : dict
+        Dictionary with the command line arguments.
+    db_config : dict
+        Database configuration.
+    rotate_angle : float
```
Comment (tobiaskleiner) [inline simtools/applications/plot_array_layout.py:273]: `rotation_angle` to be consistent with previous naming

File: simtools/applications/plot_array_layout.py [hunk 2]
```diff
@@ -122,106 +182,267 @@ def _get_site_from_telescope_list_name(telescope_list_file):
...
+    """
+    Read positions for a list of array elements from the database and return a list of layouts.
+
+    Parameters
+    ----------
+    args_dict : dict
+        Dictionary with the command line arguments.
+    db_config : dict
+        Database configuration.
+    rotate_angle : float
```
Comment (tobiaskleiner) [inline simtools/applications/plot_array_layout.py:320]: same as above

File: simtools/applications/plot_array_layout.py [hunk 3]
```diff
@@ -122,106 +182,267 @@ def _get_site_from_telescope_list_name(telescope_list_file):
...
+    """
+    Read array elements and their positions from data base using the layout name.
+
+    Parameters
+    ----------
+    args_dict : dict
+        Dictionary with the command line arguments.
+    db_config : dict
+        Database configuration.
+    rotate_angle : float
```
Comment (tobiaskleiner) [inline simtools/applications/plot_array_layout.py:365]: also here

File: simtools/applications/plot_array_layout.py [hunk 4]
```diff
@@ -122,106 +182,267 @@ def _get_site_from_telescope_list_name(telescope_list_file):
...
+    args_dict, db_config = _parse(
+        label,
+        "Plots array layout.",
+        "python applications/plot_array_layout.py --array_layout_name test_layout",
+    )
+    logger = logging.getLogger()
+    logger.setLevel(gen.get_log_level_from_user(args_dict["log_level"]))
+    io_handler_instance = io_handler.IOHandler()
+
+    rotate_angle = (
```
Comment (tobiaskleiner) [inline simtools/applications/plot_array_layout.py:410]: and here

File: simtools/model/array_model.py [hunk 5]
```diff
@@ -136,43 +147,33 @@ def site(self):
...
     def _set_config_file_directory(self):
-        """
-        Define and create config file directory.
-
-        """
+        """Define and create config file directory."""
         self._config_file_directory = self.io_handler.get_output_directory(self.label, "model")
 
-    def _build_array_model(self, site, parameters_to_change=None):
+    def _build_array_model(self, site_model, array_elements, parameters_to_change=None):
```
Comment (tobiaskleiner) [inline simtools/model/array_model.py:153]: maybe you can rename the function and adjust the description here, as it now only returns telescope model and `site_model` is already used as input.

File: simtools/visualization/visualize.py [hunk 6]
```diff
@@ -603,36 +607,39 @@ def plot_array(telescopes, rotate_angle=0, show_tel_label=False):
...
-    else:
+    pos_x_rotated, pos_y_rotated = 0.0 * u.m, 0.0 * u.m
+    if "position_x" in telescopes.colnames and "position_y" in telescopes.colnames:
         pos_x_rotated, pos_y_rotated = telescopes["position_x"], telescopes["position_y"]
-
-    pos_x_rotated, pos_y_rotated = transf.rotate(pos_x_rotated, pos_y_rotated, 90 * u.deg)
+        rotate_angle = rotate_angle + 90.0 * u.deg
+    elif "utm_east" in telescopes.colnames and "utm_north" in telescopes.colnames:
+        pos_x_rotated, pos_y_rotated = telescopes["utm_east"], telescopes["utm_north"]
+    if rotate_angle != 0:
```
Comment (tobiaskleiner) [inline simtools/visualization/visualize.py:639]: Also here I suggest to use `rotation_angle` consistently.

File: tests/integration_tests/config/plot_array_layout_one_file.yml [hunk 7]
```diff
@@ -2,9 +2,9 @@ CTA_SIMPIPE:
...
   TEST_NAME: one_file
   CONFIGURATION:
-    TELESCOPE_LIST: data/layout/telescope_positions-North-TestLayout.ecsv
+    ARRAY_LAYOUT_FILE: data/layout/telescope_positions-North-test_layout.ecsv
     MODEL_VERSION: prod6
     OUTPUT_PATH: simtools-tests
   INTEGRATION_TESTS:
-    - OUTPUT_FILE: application-plots/plot_array_layout_telescope_positions-North-TestLayout_0deg.png
-    - OUTPUT_FILE: application-plots/plot_array_layout_telescope_positions-North-TestLayout_0deg.pdf
+    - OUTPUT_FILE: application-plots/array_layout_telescope_positions-North-test_layout_North_ground_0deg.pdf
```
Comment (tobiaskleiner) [inline tests/integration_tests/config/plot_array_layout_one_file.yml:9]: should it be png here?

File: tests/integration_tests/config/plot_array_layout_one_file.yml [hunk 8]
```diff
@@ -2,9 +2,9 @@ CTA_SIMPIPE:
...
+    - OUTPUT_FILE: application-plots/array_layout_telescope_positions-North-test_layout_North_ground_0deg.pdf
```
Comment (tobiaskleiner) [inline tests/integration_tests/config/plot_array_layout_one_file.yml:10]: Should be png here

**Review by tobiaskleiner — COMMENTED** [review]
Thanks Gernot, looks good in general. I have added a few minor comments.

File: simtools/applications/plot_array_layout.py (hunk 1 above)
Comment (GernotMaier) [reply simtools/applications/plot_array_layout.py:273]: I've looked now through simtools and we are using almost everywhere `rotate_angle`. I therefore changed all `rotation_angle` to `rotate_angle`.

File: simtools/applications/plot_array_layout.py (hunk 2 above)
Comment (GernotMaier) [reply simtools/applications/plot_array_layout.py:320]: see comment above

File: simtools/applications/plot_array_layout.py (hunk 3 above)
Comment (GernotMaier) [reply simtools/applications/plot_array_layout.py:365]: see comment above.

File: simtools/applications/plot_array_layout.py (hunk 4 above)
Comment (GernotMaier) [reply simtools/applications/plot_array_layout.py:410]: see comment above.

File: simtools/model/array_model.py (hunk 5 above)
Comment (GernotMaier) [reply simtools/model/array_model.py:153]: Good point, changed it to `_build_telescope_models`.

File: simtools/visualization/visualize.py (hunk 6 above)
Comment (GernotMaier) [reply simtools/visualization/visualize.py:639]: See above.

File: tests/integration_tests/config/plot_array_layout_one_file.yml (hunk 8 above)
Comment (GernotMaier) [reply tests/integration_tests/config/plot_array_layout_one_file.yml:10]: fixed.

**Comment by GernotMaier** [comment]
@tobiaskleiner - thanks a lot for the quick review. I fixed all your suggestions, let me know if there is anything else (other just approve).

**Review by tobiaskleiner — APPROVED** [review]
All good now, thanks for the fixes.

**Pull request merged**